File: backend/app/schemas/generation.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.peptide import PeptideCandidateOut


class GenerationRunBase(BaseModel):
    task_id: Optional[int] = None
    mode: Optional[str] = None
    backend: Optional[str] = None
    count: int = 0
    min_length: Optional[int] = None
    max_length: Optional[int] = None
    temperature: Optional[float] = None
    top_p: Optional[float] = None
    status: str = "PENDING"


class GenerationRunCreate(BaseModel):
    mode: str = "Sequence-based"
    backend: str = Field(default="LOCAL_DEMO", pattern="^(LOCAL_DEMO|LOCAL_REAL_SMOKE|SERVER_PRODUCTION)$")
    count: int = 5
    min_length: Optional[int] = 15
    max_length: Optional[int] = 35
    temperature: Optional[float] = 1.0
    top_p: Optional[float] = 0.95


class GenerationRunOut(GenerationRunBase):
    id: int
    created_at: datetime
    completed_at: Optional[datetime] = None
    model_config = ConfigDict(from_attributes=True)


class GenerationRunDetailOut(GenerationRunOut):
    peptides: List[PeptideCandidateOut] = []
    disclaimer: str = ""

File: app/src/api/generation.ts
import { apiClient } from './client'
import type { PeptideCandidate } from './peptides'

export interface GenerationRun {
  id: number
  task_id?: number
  mode?: string
  backend?: string
  count: number
  min_length?: number
  max_length?: number
  temperature?: number
  top_p?: number
  status: string
  created_at: string
  completed_at?: string
}

export interface GenerationRunCreate {
  mode?: string
  backend?: 'SERVER_PRODUCTION'
  count?: number
  min_length?: number
  max_length?: number
  temperature?: number
  top_p?: number
}

export function listGenerationRuns(): Promise<GenerationRun[]> {
  return apiClient.get<GenerationRun[]>('/v1/generation-runs')
}

export function createGenerationRun(payload: GenerationRunCreate): Promise<GenerationRun> {
  return apiClient.post<GenerationRun>('/v1/generation-runs', payload)
}

export function getGenerationRun(runId: number): Promise<GenerationRun> {
  return apiClient.get<GenerationRun>(`/v1/generation-runs/${runId}`)
}

export function getGenerationRunPeptides(runId: number): Promise<GenerationRun & { peptides: PeptideCandidate[]; disclaimer: string }> {
  return apiClient.get(`/v1/generation-runs/${runId}/peptides`)
}

export interface ArtifactFile {
  name: string
  exists: boolean
  size_kb: number
  modified_at: string
  type: string
}

export interface GenerationRunArtifacts {
  artifact_dir: string | null
  files: ArtifactFile[]
  message: string
}

export function getGenerationRunArtifacts(runId: number): Promise<GenerationRunArtifacts> {
  return apiClient.get(`/v1/generation-runs/${runId}/artifacts`)
}
